### PR TITLE
Register

### PR DESCRIPTION
**Pull request by faridatarek**

### CODE REVIEW

File: lib/features/auth/Register/presentation/register_view.dart
Comment (OmarAbdelgilil) [inline lib/features/auth/Register/presentation/register_view.dart:1]: It's not a good approach to pop then push context instead use 
Navigator.of(context) .pushNamedAndRemoveUntil(your route, (Route<dynamic> route) => false);
Which will remove all previous and put the new route

Comment (OmarAbdelgilil) [reply lib/features/auth/Register/presentation/register_view.dart:1]: listenWhen: (previous, current) { return current is LoadingState || current is ErrorState || current is SuccessState; },
What is this for ?

Comment (faridatarek) [reply lib/features/auth/Register/presentation/register_view.dart:1]: i used it to compare the previous and current states to ensures the listener acts only when specific states +the code fail without it

**Review by Abdalla-Eldaly — APPROVED** [review]
good job!

**Review by OmarAbdelgilil — APPROVED** [review]

**Pull request merged**